phpBB/phpbb/profilefields/manager.php

```diff
@@ -499,4 +515,170 @@ public function build_insert_sql_array($cp_data)
 
 		return $cp_data;
 	}
+
+	/**
+	* Disable all profile fields of a certain type
+	*
+	* This should be called when an extension which has profile field types
+	* is disabled so that all those profile fields are hidden and do not
+	* cause errors
+	*
+	* @param string $profilefield_type_name Type identifier of the profile fields
+	*/
+	public function disable_profilefields($profilefield_type_name)
+	{
+		// Get the list of active profile fields of this type
+		$profile_fields = $this->list_profilefields($profilefield_type_name, true);
+
+		// If no profile fields affected, then nothing to do
+		if (!sizeof($profile_fields))
+		{
+			return;
+		}
+
+		// Update the affected profile fields to "inactive"
+		$sql = 'UPDATE ' . PROFILE_FIELDS_TABLE . '
+			SET field_active = 0
+			WHERE field_active = 1
+				AND ' . $this->db->sql_in_set('field_id', array_keys($profile_fields));
+		$this->db->sql_query($sql);
+
+		// Save modified information into a config_text field to recover on enable
+		$this->config_text->set($profilefield_type_name . '.saved', json_encode($profile_fields));
+
+		// Log activity
+		foreach ($profile_fields as $field_ident)
+		{
+			add_log('admin', 'LOG_PROFILE_FIELD_DEACTIVATE', $field_ident);
```

Comment:
We could remove the use of this deprecated function, and use the log service directly.  We would need to inject the log in the constructor, and then use something like:
`			$this->log->add('admin', $this->user->data['user_id'], (empty($this->user->ip)) ? '' : $this->user->ip, 'LOG_PROFILE_FIELD_DEACTIVATE', time(), array($field_ident));`